File: packages/shared/src/types/concept.ts
import type { ConceptId, WorkspaceId, VaultNoteId, SourceId } from "./ids.js";

export interface Concept {
  readonly id: ConceptId;
  readonly workspaceId: WorkspaceId;
  readonly name: string;
  readonly slug: string;
  readonly summary: string | null;
  readonly vaultNoteId: VaultNoteId | null;
  readonly sourceIds: readonly SourceId[];
  readonly generatedAt: string | null;
  readonly model: string | null;
  readonly createdAt: string;
  readonly updatedAt: string;
}

File: services/api/src/atlas_api/models/concept.py
"""Concept domain models — mirrors packages/shared/src/types/concept.ts."""

from pydantic import BaseModel, ConfigDict


class Concept(BaseModel):
    model_config = ConfigDict(frozen=True)

    id: str
    workspace_id: str
    name: str
    slug: str
    summary: str | None
    vault_note_id: str | None
    source_ids: tuple[str, ...]
    generated_at: str | None
    model: str | None
    created_at: str
    updated_at: str
